Phoenix/PHShebangPreprocessor.m

```diff
@@ -67,7 +67,11 @@ + (NSString *) process:(NSString *)script atPath:(NSString *)path error:(NSError
     NSPipe *standardError = [NSPipe pipe];
     NSFileHandle *standardOutputFile = standardOutput.fileHandleForReading;
 
-    task.launchPath = @"/bin/bash";
+    // Make sure we use the current user's $SHELL
+    NSDictionary *environmentDict = [[NSProcessInfo processInfo] environment];
+    NSString *shellString = [environmentDict objectForKey:@"SHELL"];
+
+    task.launchPath = shellString;
     task.standardOutput = standardOutput;
     task.standardError = standardError;
     task.arguments = @[ @"-cl", [NSString stringWithFormat:@"%@ %@", command, path] ];
```

Comment:
I suppose if issues arise with people using very esoteric shells we could add a whitelist of supported shells.  However, I think that's a pretty extreme edge case and not really worth doing for now, all things considered.